content/Watsonx/WatsonxData/106.md

```diff
@@ -0,0 +1,202 @@
+
+  ![](./images/106/data-manager-refresh-error.png)
+
+
+  You now have copies of the Great Outdoors company sales tables in object storage (hive_data), Db2 (db2catalog), and PostgreSQL (pgcatalog). This isn’t something you’d have in a real-world scenario (after all, the benefit of being able to federate access to multiple data sources is to avoid data duplication). However, it’s being done here for the purposes of highlighting Presto’s federation capabilities.
+
+  You will now see how you can run a federated query that combines data from all three of these data sources.
+
+12. Select the **Query workspace** icon from the left-side menu.
+
+13. Copy and paste the query below into the **SQL worksheet**. Click Run on **presto-01**.
+
+  This sample query could be used by the fictional business to determine which purchasing method is associated with the largest orders. The query accesses five tables, one of which is in PostgreSQL, two are in Db2, and two are in watsonx.data’s object storage.
+
+  ```bash
+    select pll.product_line_en as product, md.order_method_en as order_method, sum(sf.quantity) as total 
+    from 
+    pgcatalog.gosalesdw.sls_order_method_dim as md, 
+    db2catalog.gosalesdw.sls_product_dim as pd, 
+    db2catalog.gosalesdw.sls_product_line_lookup as pll,
+    hive_data.gosalesdw.sls_product_brand_lookup as pbl,
+    hive_data.gosalesdw.sls_sales_fact as sf 
+    where
+    pd.product_key = sf.product_key
+    and md.order_method_key = sf.order_method_key
+    and pll.product_line_code = pd.product_line_code 
+    and pbl.product_brand_code = pd.product_brand_code
+    group by pll.product_line_en, md.order_method_en 
+    order by product, order_method;
+  ```
+
+  > **Note:** If the SQL statement pastes into the worksheet as a single line, you can nicely format it by clicking the **Format worksheet** icon.
+
+  ![](./images/106/query-format.png)
+
+  The **Result set** for the query is found at the bottom of the screen.
+
+  ![](./images/106/query-result.png)
+
+14. Click the **Explain** button in the menu bar of the worksheet. 
+
+  ![](./images/106/query-result-explain.png)
+
+  The visual explain output for this query looks a lot more interesting than the one shown earlier. If you scroll through the visual explain output, you’ll see five **ScanProject** leaf nodes in the tree. These correspond to the five tables being read.
+
+  ![](./images/106/query-result-tables.png)
+
+15. Click the **X** in the upper-right corner of the **Explain** window to close it.
+
+**Optional step :** Here are two other queries you can try running as well, which combine datafrom the same three data sources.
+
+1. The following query displays all Canadian and Mexican employees, along with their region and country. This is the kind of query that a reporting tool might generate, based on input from the user.
+
+  ```bash
```

Comment:
Is there a way we could indent this code so it appears more like it does in the original lab guide?

Reply:
fixed alignment